File: search/types/whats-on.ts
export type WhatsOnSource = {
  title: string
  description: string
  end_date: Date
  start_date: Date
  image_url: string
  image_alt: string
  format: string
}

export type WhatsOnHit = {
  _id: string
  _score: string
  _source: WhatsOnSource
}

export type WhatsOn = {
  id: string
  title: string
  description: string
  end_date: Date
  start_date: Date
  image_url: string
  image_alt: string
  format: string
}

File: pipeline/src/graph/models/whats_on.py
from neomodel import (
    DateProperty,
    StringProperty,
    StructuredNode,
    UniqueIdProperty,
)


class Exhibition(StructuredNode):
    uid = UniqueIdProperty()
    format = StringProperty()
    title = StringProperty(required=True)
    description = StringProperty()
    start_date = DateProperty()
    end_date = DateProperty()
    image_url = StringProperty()
    image_alt = StringProperty()
    location = StringProperty()


class Event(StructuredNode):
    uid = UniqueIdProperty()
    format = StringProperty()
    title = StringProperty(required=True)
    description = StringProperty()
    start_date = DateProperty()
    end_date = DateProperty()
    image_url = StringProperty()
    image_alt = StringProperty()
    location = StringProperty()
